python/ray/serve/_private/deployment_state.py

```diff
@@ -69,8 +69,9 @@ class ReplicaState(Enum):
 class ReplicaStartupStatus(Enum):
     PENDING_ALLOCATION = 1
     PENDING_INITIALIZATION = 2
-    SUCCEEDED = 3
-    FAILED = 4
+    PENDING_INITIAL_HEALTH_CHECK = 3
```

Comment:
what's the reason to separate this from `PENDING_INITIALIZATION`? if there is no need for the distinction let's err on the side of having fewer states (easier to reason about)